modules/nw-kuryr-cleanup.adoc

```diff
@@ -0,0 +1,208 @@
+// Module included in the following assemblies:
+//
+// * networking/ovn_kubernetes_network_provider/migrate-from-kuryr-sdn.adoc
+
+:_content-type: PROCEDURE
+[id="nw-kuryr-cleanup_{context}"]
+= Cleaning up resources after migration
+
+After migration is done, there are necessary steps to clean up resources used by Kuryr. From high level perspective it will include:
+
+* Removing Kuryr finalizers from resources:
+* Removing Kuryr CRDs
+* Removing {rh-openstack-first} resources
+
+.Prerequisites
+
+* Install the OpenShift CLI (`oc`).
+* Install `jq` tool
+* Install Python interpreter
+* Install `openstacksdk` python package
+* Install `openstack` cli tool
+* Access to the underlying {rh-openstack}
+* Access to the cluster as a user with the `cluster-admin` role.
+
+.Procedure
+
+. Find out cluster and Kuryr related identifiers, which will be used later:
++
+[source,terminal]
+----
+$ CLUSTERID=$(oc get infrastructure.config.openshift.io cluster -o=jsonpath='{.status.infrastructureName}')
+$ CLUSTERTAG="openshiftClusterID=${CLUSTERID}"
+$ ROUTERID=$(oc get kuryrnetwork -A --no-headers -o custom-columns=":status.routerId"|head -n 1)
+----
+
+. Create helper bash function, which will help out with removing finalizers
+from specified resources. It might look complicated, but it simply gets two
+parameters - first one is name of the resource, and the second is the
+finalizer to be removed. Than it will get all the desired resource from the
+cluster and replace it's definition with the same data excluding specified
+finalizer.
++
+[source,terminal]
+----
+$ function REMFIN {
+    local resource=$1
+    local finalizer=$2
+    for res in $(oc get $resource -A --template='{{range $i,$p := .items}}{{ $p.metadata.name }}|{{ $p.metadata.namespace }}{{"\n"}}{{end}}'); do
+        name=${res%%|*}
+        ns=${res##*|}
+        oc get -n $ns $resource $name -o json |
+            jq -Mcr "if .metadata.finalizers != null then del(.metadata.finalizers[] | select(. == \"${finalizer}\")) else . end" |
+            oc replace -n $ns $resource $name -f -
+    done
+}
+----
+. Remove Kuryr finalizers from services:
++
+[source,terminal]
+----
```

Comment:
should we follow the same pattern Kuryr follows of first deleting the openstack resource, then removing finalizer for kuryr CR and then finalizer from native Kubernetes resource?

Reply:
The process is designed with 2 assumptions:
* Minimize the effect on core Kubernetes resources (mainly allow Pods and Services to be deleted; Namespaces need to wait a bit more) ASAP.
* Only delete CRDs once we deleted all the OpenStack resources these CRDs are necessary to identify. E.g. we don't care about KuryrPorts too much because we can delete ports by networks. Similarly even if Service is gone we can delete LBs using tags.

That's why it's not following the same pattern as in kuryr-controller.